### PR TITLE
feat(APIv2): add dashboard permissions in the current parameters api endpoint

### PR DESCRIPTION
**Pull request by abuathier**
## Description

Jira: 🏷️ **MON-19376**

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [X] 23.10.x (master)

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [X] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [X] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/User/Infrastructure/API/FindCurrentUserParameters/FindCurrentUserParametersPresenter.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                'locale' => $data->locale,
+                'is_admin' => $data->isAdmin,
+                'use_deprecated_pages' => $data->useDeprecatedPages,
+                'is_export_button_enabled' => $data->isExportButtonEnabled,
+                'theme' => $data->theme->value,
+                'user_interface_density' => $data->userInterfaceDensity->value,
+                'default_page' => $data->defaultPage,
+            ];
+
+            if ($this->hasDashboardFlag && $data->dashboardPermissions->globalRole) {
```
Comment (abuathier) [inline centreon/src/Core/User/Infrastructure/API/FindCurrentUserParameters/FindCurrentUserParametersPresenter.php:65]: Output a `null` when the feature flag is ON.

File: centreon/tests/api/features/ConfigurationUsers.OpenApiDoc.feature [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    ACLGROUP;add;name-creator-ACLGROUP;alias-creator-ACLGROUP
+    ACLGROUP;addmenu;name-creator-ACLGROUP;name-creator-ACLMENU
+    ACLGROUP;setcontact;name-creator-ACLGROUP;usr-creator;
+    """
+    Given I am logged in with "usr-creator"/"Centreon@2023"
+
+    When I send a GET request to '/api/latest/configuration/users/current/parameters'
+    Then the response code should be "200"
+    And the JSON node "name" should be equal to "usr-creator"
+    And the JSON node "dashboard" should not exist
```
Comment (abuathier) [inline centreon/tests/api/features/ConfigurationUsers.OpenApiDoc.feature:107]: Change to test `null`

File: centreon/tests/api/features/ConfigurationUsers.OpenApiDoc.feature [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    """
+    CONTACT;ADD;usr-creator;usr-creator;usr-creator@centreon.test;Centreon@2023;0;1;en_US;local
+    CONTACT;setparam;usr-creator;reach_api;1
+    ACLMENU;add;name-creator-ACLMENU;alias-creator-ACLMENU
+    ACLMENU;grantrw;name-creator-ACLMENU;0;Home;Dashboard;Creator;
+    ACLGROUP;add;name-creator-ACLGROUP;alias-creator-ACLGROUP
+    ACLGROUP;addmenu;name-creator-ACLGROUP;name-creator-ACLMENU
+    ACLGROUP;setcontact;name-creator-ACLGROUP;usr-creator;
+    """
+    Given I am logged in with "usr-creator"/"Centreon@2023"
```
Comment (abuathier) [inline centreon/tests/api/features/ConfigurationUsers.OpenApiDoc.feature:102]:
```suggestion
    Given I am logged in with "usr-creator"/"Centreon@2023"
    And a feature flag "dashboard" of bitmask 0
```

File: centreon/tests/api/features/ConfigurationUsers.JsonSchema.feature [hunk 4]
```diff
@@ -0,0 +1,25 @@
+Feature:
```
Comment (callapa) [inline centreon/tests/api/features/ConfigurationUsers.JsonSchema.feature:1]: All feature files are in CamelCase and are not separated by dots.

File: centreon/tests/api/features/ConfigurationUsers.OpenApiDoc.feature [hunk 5]
```diff
@@ -0,0 +1,150 @@
+Feature:
```
Comment (callapa) [inline centreon/tests/api/features/ConfigurationUsers.OpenApiDoc.feature:1]: All feature files are in CamelCase and are not separated by dots.

**Review by a-launois — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**